src/Nethermind/Nethermind.Trie/NodeStorage.cs

```diff
@@ -119,11 +119,11 @@ private static Span<byte> GetHashBasedStoragePath(Span<byte> pathSpan, in ValueH
 
         if (Scheme == INodeStorage.KeyScheme.HalfPath && (readFlags & ReadFlags.HintReadAhead) != 0)
         {
-            if (address is null && path.Length > TopStateBoundary)
+            if (address == default && path.Length > TopStateBoundary)
```

Comment:
as comparing to default is now expensive, maybe lets save that to variable and avoid 2nd comparison in else